nats.go

```diff
@@ -2810,7 +2810,7 @@ func (nc *Conn) processPong() {
 	nc.mu.Lock()
 	if len(nc.pongs) > 0 {
 		ch = nc.pongs[0]
-		nc.pongs = nc.pongs[1:]
+		nc.pongs = append(nc.pongs[:0], nc.pongs[1:]...)
```

Comment:
Any issue specifically that this is addressing? Is that to prevent reallocating/expanding the slice?

Reply:
Yes this pattern which I have used I realized with JS work is unbounded growth..

Reply:
before this change noticed it would be reallocating and expanding the capacity each time, so len + cap on the array would have look like this:

```
len cap
245 280
244 279
243 278
242 277
241 276
240 275
239 274
```

and with this change the capacity grows like this instead:

```
len cap
10 16
14 16
16 16
...
29 32
32 32
31 32
...
43 64
42 64
41 64
...
```

Reply:
I tried to search in our code base if we do that in other places (which I thought we were doing) but could not find. Anyway, let's keep an eye on that pattern and fix if we stumble on it.